### ISSUE TITLE
Cannot read property 'subTree' of undefined if expose() is used

### ISSUE THREAD

**Issue by KaelWD** [vuejs/test-utils#435]
```js
it('works with expose', async () => {
  const Component = defineComponent({
    setup (props, { expose }) {
      const val = ref('')

      expose({
        val
      })

      return () => h('div')
    }
  })

  const wrapper = mount(Component)
})
```

In `VueWrapper.hasMultipleRoots`, `this.vm` is the exposed object instead of the component instance. 

https://github.com/vuejs/vue-test-utils-next/blob/c667907bac0db961439ccdca0644b54a5f5ee382/src/mount.ts#L437-L449

Should be something like this but always returning `i.proxy`:

https://github.com/vuejs/vue-next/blob/fbf2d68bbcde75c95363e1fd17946c720d0f3925/packages/runtime-core/src/componentPublicInstance.ts#L207-L218

**Comment by lmiller1990** [comment]
Thanks for the bug report and detective work. If you don't get a change to dig into this more, I will try get a fix done sooner rather than later. Have not played with `expose` much (yet).

**Comment by KaelWD** [comment]
I think this is actually vue's fault, if `expose` is used it doesn't expose the component's proxy at all:  
https://github.com/vuejs/vue-next/commit/0e59770b9282992f6a5af4d8fef33dafb948fc8b#diff-56930a63772d2e89f19432918b9d6ed851ede4159b0d32404e0c1d8c159fec99R313-R315

**Comment by freakzlike** [comment]
@KaelWD Can you verify this again with `rc.21`? There have been some fixes and changes (#1067 and https://github.com/vuejs/test-utils/commit/e80b7b28046362fb1e5a281353637064722fbc6b) and also a unit test with this case already exists.

**Comment by lmiller1990** [comment]
Looks like we do have a unit test - let's close this. We can re-open if it's not actually fixed.
